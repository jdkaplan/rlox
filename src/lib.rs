macro_rules! debug_log_gc {
    ($($arg:tt)*) => {{
        #[cfg(feature = "log_gc")]
        println!($($arg)*);
    }};
}

mod chunk;
mod compiler;
mod gc;
mod heap;
mod object;
mod scanner;
mod table;
mod value;
mod vm;

pub use vm::{InterpretError, InterpretResult, Vm};

pub const U8_COUNT: usize = (u8::MAX as usize) + 1;
pub const FRAMES_MAX: usize = 64;
pub const STACK_MAX: usize = FRAMES_MAX * U8_COUNT;
