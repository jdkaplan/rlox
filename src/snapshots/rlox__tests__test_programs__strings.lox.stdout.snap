---
source: src/lib.rs
description: "var s = \"\";\nprint s == \"\";\n\nprint s; s = s + \"abcd\";\nprint s; s = s + \"5678\";\nprint s; s = s + \"efgh\";\nprint s; s = s + \"9012\";\nprint s;\n\nprint s == \"abcd5678efgh9012\";\n"
input_file: test_programs/strings.lox
---
true

abcd
abcd5678
abcd5678efgh
abcd5678efgh9012
true

